### PR TITLE
Use `noarch: python` for Python packages

### PR DESCRIPTION
**Pull request by hcho3**
Closes #128

As the XGBoost Python packages are pure Python (using `ctypes` to interact with the `libxgboost` library), there is no need to build these for every Python version and architecture. So simply mark these as `noarch: python` to get the minimum needed builds and move on.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by hcho3** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Comment by xhochy** [comment]
The errors could be better explained by switching to `conda-build` here.

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by xhochy** [comment]
Found the issue in the logs: `Unsatisfiable dependencies for platform linux-64`. It tries to build the `noarch` packages on the `build_platform` but in the cross-compile setting we don't have the native packages available for that yet.

**Comment by xhochy** [comment]
This can be fixed by:

* Switch dependencies of the `noarch` packages from `exact=True` to `x.x.x`
* Skip them except for `linux-64`

**Comment by jakirkham** [comment]
Thanks Uwe! 🙏

Was wondering if that might be the case

Have done the first step of relaxing the pin of `libxgboost`. Not sure if we will need to do this elsewhere or not

Think we may need to skip other platforms as you suggest (and maybe other CUDA versions), but want to see how things behave with the first change before changing this

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

**Comment by jakirkham** [comment]
It looks like most things passed except the macOS jobs had some issue installing Conda

Maybe this was due to some other conda-forge infrastructure issues that occurred at the time

Going to restart CI to see if that fixes the issue (and confirm the other jobs are still working ok)

**Comment by jakirkham** [comment]
@conda-forge-admin , please restart CI

**Comment by jakirkham** [comment]
Looks like we are seeing the same install error [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=847443&view=logs&jobId=ced5d8de-8227-5f3f-33a1-45cf1592c45a&j=ced5d8de-8227-5f3f-33a1-45cf1592c45a&t=f7bcb349-6dca-53ba-d88d-1d9963e119ba )

```
LibMambaUnsatisfiableError: Encountered problems while solving:
  - package conda-forge-ci-setup-4.3.1-py310h84be057_100 requires conda >=23.9, but none of the providers can be installed
```

File: conda-forge.yml [hunk 1]
```diff
@@ -6,7 +6,10 @@ build_platform:
   osx_arm64: osx_64
 conda_build:
   pkg_format: '2'
+conda_build_tool: conda-build
 conda_forge_output_validation: true
+conda_install_tool: conda
+conda_solver: libmamba
```
Comment (jakirkham) [inline conda-forge.yml:12]: Should we configure these differently?

Comment (jaimergp) [reply conda-forge.yml:12]: Try with `conda_install_tool: mamba`

Comment (jakirkham) [reply conda-forge.yml:12]:
```suggestion
conda_build_tool: conda-build
conda_forge_output_validation: true
conda_install_tool: mamba
conda_solver: libmamba
```

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
The re-rendering bot is having an issue related to the recent `conda-build` release ( https://github.com/conda/conda-build/issues/5125 ). There's already some work to pin things down in conda-forge ( https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/621 ). To fix this am [redeploying the webservice action]( https://github.com/conda-forge/webservices-dispatch-action/actions/runs/7282473509/job/19844889548 )

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
It's still pushing the image. Will try back in a bit

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
Ok it's all cleared up. Latest [re-rendering bot log]( https://github.com/conda-forge/xgboost-feedstock/actions/runs/7282604412 )

**Comment by jakirkham** [comment]
There are a bunch of CUDA bits in the `noarch` packages. Wondering if maybe we can push these back into `libxgboost` to simplify this a bit

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Pull request merged**